pkg/arvo/neo/cod/std/src/fil/diary.md

```diff
@@ -0,0 +1,297 @@
+  ::  lads:neo is (map pish:neo lash:neo)
+  %-  ~(gas by *lads:neo)
+  :~  :-  ::
+          ::  pish:neo
+          ::  to simplify: [%.n @da] means the kid's
+          ::  path contains any @da, and %.n is there
+          ::  to signify that the pith can not have more
+          ::  fields afterwards
+          [[%.n %da] %.n]
+      ::
+      ::  lash:neo is (pair curb:neo (set stud:neo))
+      ::  curb:neo defines the kids' state
+      ::  (set stud:neo) defines the kids' pokes
+      [[%only %txt] ~]
+  ==
+```
+
+In this case, Diary only constrains its immediate children. Each of its children lives at `/path/to/diary/<@da>`, where `<@da>` is used as an ID for the entry, and those children are just `%txt`s (cords) which store the text of the diary entry.
+
+There’s no need for Diary to recusrively constrain every single path that nests under `/path/to/diary`, since the app only needs to reserve the first “generation” of shrubs below it in order to work. In this case, the developer has forbidden any shrubs being created under the diary entries by ending the `pish` in a `%.n`. If that were `%.y`, any app could make a shrub at `/path/to/diary/<@da>/comments`. or `/path/to/diary/<@da>/backlinks`.
+
+To define the shrub’s kids, we create an empty map which is the bunt of `lads:neo`, then use `+gas:by` to populate the map with our own data, which will be a `(list (pair pish:neo lash:neo))`.
+
+The `pish:neo` statically types the paths which we’ll allow to be created beneath this shrub, and the `lash:neo` defines the kids’ state and pokes. In this case, the kids take no pokes and their state can only be `%txt`.
+
+(It’s worth flagging that `$curb:neo` contains several combinatorial rules about state types. For example `[%pro %txt]` would mean “the state of the child shrub can be any type which is readily convertible into a `%txt` with a conversion we have available in the `/con` folder”. This gives us a clue as to how we could handle state transitions and interoperability over the lifetime of our shrub, but it’s outside the remit of this tutorial.)
+## bowl:neo
+Notice that the `src` in `bowl:neo` differs from `bowl:gall`. Here’s the new type in full.
+
+```hoon
++$  bowl
+  $:  src=[=ship =pith]                 ::  a request's source ship and shrub
+      our=@p                            ::  our ship
+      here=pith                         ::  path to this shrub
+      now=@da                           ::  current datetime
+      eny=@uvJ                          ::  entropy
+      deps=(map term (pair pith lore))  ::  dependencies
+      kids=lore                         ::  XX
+  ==
+```
+
+## Generating cards, tombstoning shrubs
+We covered `card:neo` in the Counter tutorial, but this is the first time we’re seeing one generated within a shrub. Diary takes two pokes `%put-entry`, to create a new diary entry, and `%del-entry` to tombstone one.
+
+Here’s the `+poke` arm of the Diary shrub, expanded with comments.
+
+```hoon
+++  poke
```

Comment:
The Counter tutorial had filenames at the top of code blocks. I saw those while looking at NEAR docs and quite liked the idea; should commit to including them everywhere or nowhere.